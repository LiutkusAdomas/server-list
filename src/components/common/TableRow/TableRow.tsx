import { Server } from '../../../model/Server.type';

type TableRowPropsType = {
  item: Server;
};

export const TableRow: React.FC<TableRowPropsType> = ({ item }) => {
  return (
    <tr className="border-b">
      <td className="py-2 px-4">
        <img
          src={item.flag}
          alt={item.name}
          className="shadow-lg m-1"
          style={{ maxWidth: '80px', minWidth: '40px' }}
        />
      </td>
      <td className="py-2 px-4 text-center">{item.name}</td>
      <td className="py-2 px-4 text-center">{item.distance}</td>
    </tr>
  );
};
