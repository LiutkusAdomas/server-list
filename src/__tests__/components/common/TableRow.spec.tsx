import { render } from '@testing-library/react';
import { describe, expect, it } from 'vitest';
import { TableRow } from '../../../components/common/TableRow/TableRow';
import { Server } from '../../../model/Server.type';

const server: Server = {
  id: '1',
  name: 'Test',
  distance: 1,
  flag: 'flag.svg',
};

describe('Table Row component', () => {
  it('should render a table row component', () => {
    const { asFragment } = render(<TableRow item={server} />);
    expect(asFragment).toMatchSnapshot();
  });
});
