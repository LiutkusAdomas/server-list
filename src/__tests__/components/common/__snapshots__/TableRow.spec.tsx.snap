// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Table Row component > should render a table row component 1`] = `[Function]`;
